pkg/app/web/src/components/application-form/index.tsx

```diff
@@ -718,10 +717,13 @@ const UnregisteredApplicationList: FC<ApplicationFormProps> = memo(
                           color="primary"
                           type="submit"
                           onClick={() => {
+                            const envId = envsMap.has(app.envName)
+                              ? (envsMap.get(app.envName) as string)
+                              : "";
                             // NOTE: Repo remote and branch aren't needed because they are populated by API.
                             setAppToAdd({
                               name: app.name,
-                              env: envsMap.get(app.envName) as string,
+                              env: envId,
```

Comment:
nits,
```suggestion
                              env: envsMap.get(app.envName) | "",
```
and the `envId` defined in L740 can be removed.